### DISCUSSION TITLE
Warnings for web? Expected way on web?

### DISCUSSION THREAD

**Discussion by vietstone-ng** [cfug/dio#2306] [General]
I see these warnings in file:`options.dart`, function:`compose` for web.
```
    if (data != null && kIsWeb) {
      if (sendTimeout != null && sendTimeout! > Duration.zero) {
        warningLog(
          'sendTimeout cannot be used without a request body to send on Web',
          StackTrace.current,
        );
      }
      if (onSendProgress != null) {
        warningLog(
          'onSendProgress cannot be used without a request body to send on Web',
          StackTrace.current,
        );
      }
    }
```

It confused me what's correct way of using it on web, with or without `data`? Because if I upload a file (with data) and listen to progress, it will shows warning in console.

**Comment by AlexV525** [comment]
Ah, I guess the condition should be data == null

**Comment by kowalgregy** [comment]
Same problem, it all works, but I'm getting this warning message.
